lib/vstd/DateUtils.cpp

```diff
@@ -12,16 +11,17 @@ namespace vstd
 
 	DLL_LINKAGE std::string getFormattedDateTime(std::time_t dt)
 	{
-#if defined(VCMI_ANDROID)
-		CAndroidVMHelper vmHelper;
-		return vmHelper.callStaticStringMethod(CAndroidVMHelper::NATIVE_METHODS_DEFAULT_CLASS, "getFormattedDateTime");
-#endif
+		std::string lang = settings["general"]["language"].String();
+		std::string locale = Languages::getLanguageOptions(lang).locale;
 
 		std::tm tm = *std::localtime(&dt);
 		std::stringstream s;
 		try
 		{
-			s.imbue(std::locale(""));
+			if(locale.empty())
+				s.imbue(std::locale(""));
+			else
+				s.imbue(std::locale(locale));
```

Comment:
Using boost::locale (or replacing all this handling with boost::datetime) might be a good idea.

Problem is, c++ standard does not defines list of supported encodings. 
Whether "de_DE.UTF-8" exists and/or is usable depends on OS, OS language, OS version and phase of the moon.

For reliable, cross-platform behavior boost might be better option

Reply:
Do you have an example? I do not find anything fitting our usecase.

Only examples that still using `imbue`.

Reply:
boost locale likely uses libUnicode or some other more reliable locale generation than OS.

So imbuing locale with whatever boost locale provides might be better option. However please check for 'imbue' references in code. I think we already use boost locale somewhere to initialize global locale. But I think only in some cases?

Reply:
I don't get it working...

The result (at least for "de_DE.UTF-8" is incorrect.

```
	DLL_LINKAGE std::string getFormattedDateTime(std::time_t dt)
	{
		std::string lang = settings["general"]["language"].String();
		std::string localeCode = Languages::getLanguageOptions(lang).locale;
		boost::locale::generator localeGenerator = boost::locale::generator();
		boost::gregorian::date_facet *facet = new boost::gregorian::date_facet{"%x %X"};
		std::locale locale = std::locale(localeCode.empty() ? localeGenerator("") : localeGenerator(localeCode), facet);
		boost::posix_time::ptime tm = boost::posix_time::ptime(boost::gregorian::date(1970, 1, 1)) + boost::posix_time::seconds(static_cast<long>(dt));

		std::stringstream s;
		try
		{
			s.imbue(locale);
		}
		catch(const std::runtime_error & e)
		{
			// locale not be available - keep default / global
		}
		s << tm;
		return s.str();
	}
```

Reply:
ICU as backend gives incorrect results. If I switch to std. It's fine on PC. But not android